packages/experiments/README.md

```diff
@@ -7,33 +7,42 @@ This package acts as a "dealer" that only allows WordPress packages to use the e
 Each package needs to start by registering itself:
 
 ```js
-const { register } =
-	__dangerousOptInToUnstableAPIsOnlyForCoreModules(
-		'<CONSENT STRING>',  // See index.js, this may change without notice.
-		'@wordpress/blocks'
-	);
+const { lock, unlock } = __dangerousOptInToUnstableAPIsOnlyForCoreModules(
+	'<CONSENT STRING>', // See index.js, this may change without notice.
+	'@wordpress/blocks'
+);
 ```
 
 The function name communicates that plugins are not supposed to use it. To make double and triple sure, the first argument must be that exact consent string, and the second argument must be a known `@wordpress` package that hasn't opted in yet – otherwise an error is thrown.
 
-Expose a new `__experimental` API as follows:
+The `lock` and `unlock` API may be used as follows:
 
 ```js
-export const __experiments = register( { __unstableGetInnerBlocksProps } )
-```
-
-Consume the registered `__experimental` APIs as follows:
-
-```js
-// In the @wordpress/block-editor package:
-import { __experiments as blocksExperiments } from '@wordpress/blocks';
-const { unlock } =
-	__dangerousOptInToUnstableAPIsOnlyForCoreModules(
-		'<CONSENT STRING>',  // See index.js
-		'@wordpress/block-editor'
-	);
-
-const { __unstableGetInnerBlocksProps } = unlock( blocksExperiments );
+// In the package exposing the experimental APIs:
+const { lock } = __dangerousOptInToUnstableAPIsOnlyForCoreModules(
```

Comment:
Can you share an example with a package exporting an experimental component?

Reply:
Sure!

```js
// In @wordpress/package1/index.js:
import { lock } from './experiments';

export function DataListing({data}) {
	return <ExperimentalDataListing data={data} format='list' />;
}
function ExperimentalDataListing({data, format}) {
	if(format === 'list') {
		return (<ul> {/* ... */} </ul>)
	} else if(format === 'table') {
		return (<table> {/* ... */} </table>)
	}
}

// In @wordpress/package2/index.js:
import { DataListing } from '@wordpress/package1';
import { unlock } from './experiments';

const ExperimentalDataListing = unlock(DataListing);
export function MyComponent() {
	return (
		<ExperimentalDataListing data={data} format="table" />
	)
}
```

Without the `unlock()` function you can only use the public `DataListing` component. With the `unlock()` access you can also access the private `ExperimentalDataListing` component.

Reply:
If I am understanding the proposal correctly, there is a missing `lock( DataListing, ExperimentalDataListing )` call in your example, right?